File: Python/0851-0900/867. Transpose Matrix.py
# https://leetcode.com/problems/transpose-matrix/

class Solution:
    def transpose(self, matrix: List[List[int]]) -> List[List[int]]:
        rowSize = len(matrix)
        colSize = len(matrix[0])

        result = []

        for col in range(0, colSize):
            current = []
            for row in range(0, rowSize):
                current.append(matrix[row][col])
            
            result.append(current)

        return result

File: Javascript/0851-0900/867. Transpose Matrix.js
// https://leetcode.com/problems/transpose-matrix/

var transpose = function(matrix) {
    const rowSize = matrix.length;
    const colSize = matrix[0].length;

    result = [];

    for(let col = 0; col<colSize; col++){
        current = []
        for(let row = 0; row < rowSize; row++){
            current.push(matrix[row][col]);
        }
        result.push(current);
    }

    return result;
};
